openupgrade_scripts/scripts/account/14.0.1.1/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,648 @@
+
+account      / account.bank.statement.line / _inherits                     : NEW _inherits: {'account.move': 'move_id'}
+account      / account.bank.statement.line / move_id (many2one)            : NEW relation: account.move, required
+account      / account.move             / statement_line_id (many2one)  : NEW relation: account.bank.statement.line
+# pre-migration: added move_id column and filled with an account.move id related to the move_name field
+# TODO: in v13, an account.move was created after the reconciliation (move_name field). In v14, a temporary account.move is created when
+# the statement line is created. In cases where move_name is empty, this temporary account.move has to be created as move_id field is required
+
+account      / account.bank.statement.line / account_id (many2one)         : DEL relation: account.account
+account      / account.bank.statement.line / journal_entry_ids (one2many)  : DEL relation: account.move.line
+account      / account.bank.statement.line / bank_account_id (many2one)    : DEL relation: res.partner.bank
+account      / account.bank.statement.line / move_name (char)              : DEL
+account      / account.bank.statement.line / name (char)                   : not stored anymore
+account      / account.bank.statement.line / name (char)                   : now related
+# NOTHING TO DO: This model uses the fields from the inherited account.move model (move.id field)
+
+account      / account.bank.statement.line / amount_residual (float)       : NEW isfunction: function, stored
+# NOTHING TO DO: filled by Odoo.
+
+account      / account.bank.statement.line / foreign_currency_id (many2one): NEW relation: res.currency
+# DONE: pre-migration: This field is the "currency_id" field from v13. It has been renamed because now, "currency_id" in v14 stores
+# the "journal_currency_id" values from v13.
+
+account      / account.bank.statement.line / is_reconciled (boolean)       : NEW isfunction: function, stored
+# NOTHING TO DO: new feature. Technical field to indicate if statement line is already reconciled for display purposes.
+
+account      / account.bank.statement.line / company_id (many2one)         : not stored anymore
+account      / account.bank.statement.line / date (date)                   : not stored anymore
+account      / account.bank.statement.line / date (date)                   : now related
+account      / account.bank.statement.line / ref (char)                    : not stored anymore
+account      / account.bank.statement.line / ref (char)                    : now related
+account      / account.bank.statement.line / journal_id (many2one)         : not stored anymore
+account      / account.bank.statement.line / journal_id (many2one)         : now required
+# NOTHING TO DO: These fields now are in account.move
+
+account      / account.bank.statement.line / note (text)                   : DEL
+# NOTHING TO DO
```

Comment:
Do you think maybe the note contents can be appended to the related move's narration column? This field is displayed in statement line context: https://github.com/odoo/odoo/blob/14.0/addons/account/views/account_bank_statement_views.xml#L290

Reply:
Not sure if we have to do the same with the communication field of account.payment

Reply:
Ah, account.payment now also inherit from account.move. In that case I would say, yes.

Reply:
done